newswires/client/src/urlState.ts

```diff
@@ -70,10 +71,18 @@ export function urlToConfig(location: {
 		dateRange: { start, end },
 	};
 
-	if (page === 'feed') {
-		return { view: 'feed', query };
-	} else if (page.startsWith('item/') && page.split('/').length === 2) {
-		return { view: 'item', itemId: page.split('/')[1], query };
+	if (page.includes('feed')) {
+		return { view: 'feed', query, ticker: page.includes('ticker') };
+	} else if (
+		page.includes('item/') &&
+		(page.split('/').length === 2 || page.split('/').length === 3)
+	) {
+		return {
+			view: 'item',
+			itemId: page.split('/').pop()!,
```

Comment:
small thing, but it would be nice to avoid mutation with `pop` here I think. 

[`Array.at`](https://caniuse.com/?search=array.at) probably doesn't have enough support yet..? But given that all the branches want to match on elements of `page.split`, we could do something like:

```
const pageSegments = page.split("/")

if (pageSegments.includes("feed")) {
    ...
} (...) {
    return {
        ...
        itemId: pageSegments[pageSegments.length - 1]
        ...
    }
}
```

what do you think?

Reply:
yes, pure functions are safer.